lib/koans/18_genservers.ex

```diff
@@ -0,0 +1,68 @@
+defmodule GenServers do
+  use Koans
+
+  @intro "GenServers"
+
+  defmodule ExampleGenServer do
+    use GenServer
+
+    # Use init to set up initial state []
+    def init(_args) do
+      {:ok, []}
+    end
+
+    # Use calls to perform synchronous actions
+    def handle_call(:pop, _from, [head|tail]) do
+      {:reply, head, tail}
+    end
+
+    def handle_call(:peek, _from, stack) do
+      {:reply, stack, stack}
+    end
+
+    # Use casts to perform asynchronous actions
+    def handle_cast({:push, value}, stack) do
+      {:noreply, [value|stack]}
+    end
+  end
+
+
+  koan "You can think of GenServers like more flexible agents" do
```

Comment:
It's hard to see by the examples how GenServers are "more flexible" than Agents. Part of what is useful about GenServers is the common interface, right? I'm pulling a little bit from memory here, so forgive errors, but the interface is used by supervisors, yes?